File: 2015/golang/days/day1/day1.go
package day1

import (
	"strings"
)

// Part1 solves the first part of day 1 challenge.
func Part1(input string) int {
	floor := 0
	inputSplice := strings.Split(input, "")
	for _, value := range inputSplice {
		if value == "(" {
			floor++
		} else {
			floor--
		}
	}

	return floor
}

// Part2 solves the first part of day 1 challenge.
func Part2(input string) int {
	floor := 0
	inputSplice := strings.Split(input, "")
	for index, value := range inputSplice {
		if value == "(" {
			floor++
		} else {
			floor--
		}
		if floor < 0 {
			return index + 1
		}
	}
	return floor
}

File: 2015/c/src/day1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_LINE_LENGTH 1024

void part1(const char* filename) {
    FILE *file = fopen(filename, "r");
    if (!file) {
        fprintf(stderr, "Error: Could not open file %s\n", filename);
        return;
    }

    char line[MAX_LINE_LENGTH];
    int result = 0;

    while (fgets(line, sizeof(line), file)) {
        // Remove newline if present
        line[strcspn(line, "\n")] = 0;

        // TODO: Implement part 1 logic here
    }

    fclose(file);
    printf("Part 1: %d\n", result);
}

void part2(const char* filename) {
    FILE *file = fopen(filename, "r");
    if (!file) {
        fprintf(stderr, "Error: Could not open file %s\n", filename);
        return;
    }

    char line[MAX_LINE_LENGTH];
    int result = 0;

    while (fgets(line, sizeof(line), file)) {
        // Remove newline if present
        line[strcspn(line, "\n")] = 0;

        // TODO: Implement part 2 logic here
    }

    fclose(file);
    printf("Part 2: %d\n", result);
}

int main(int argc, char *argv[]) {
    const char *filename = (argc > 1) ? argv[1] : "input/day1.txt";

    printf("Advent of Code 2015 - Day 1\n");
    printf("Input file: %s\n\n", filename);

    part1(filename);
    part2(filename);

    return 0;
}